packages/plugins/rubygem-smart_proxy_salt/rubygem-smart_proxy_salt.spec

```diff
@@ -97,14 +102,6 @@ gem unpack %{SOURCE0}
 
 %setup -q -D -T -n  %{gem_name}-%{version}
 
-%if %{with python2} && 0%{?rhel} >= 8
-sed -i -e '1s|^#!.*$|#!%{__python2}|' sbin/upload-salt-reports
-%endif
-
-%if %{with python3} && 0%{?rhel} >= 8
-sed -i -e '1s|^#!.*$|#!%{__python3}|' sbin/upload-salt-reports
-%endif
```

Comment:
Why was the old code replaced with a shell wrapper? IMHO this is a huge step back. In packaging we already know what the Python version should be.

Reply:
With the shell wrapper, we can distinguish between the Python version when the package is installed already. This makes Py2 to Py3 migrations easier: the user won't need to reinstall Smart Proxy Salt since the wrapper just uses the version which works.
With the current solution, the package must be reinstalled to enable Python3.

What do you think about it?

Reply:
That the previous approach was the correct one. You already know what the platform uses. On EL7 you will use Python 2 regardless. On EL8 it will always be Python 3. Trying to determine it on every single request is just overhead. So for packaged installations I'd stick with what's there since it's the right way to do it.

Reply:
Implemented as suggested.